src/Type/Union/PostObjectUnion.php

```diff
@@ -1,26 +1,63 @@
 <?php
 
-namespace WPGraphQL\Type;
+namespace WPGraphQL\Type\Union;
 
-use WPGraphQL\Types;
+use WPGraphQL\Registry\TypeRegistry;
 
-$possible_types     = [];
-$allowed_post_types = \WPGraphQL::get_allowed_post_types();
-if ( ! empty( $allowed_post_types ) && is_array( $allowed_post_types ) ) {
-	foreach ( $allowed_post_types as $allowed_post_type ) {
-		if ( empty( $possible_types[ $allowed_post_type ] ) ) {
-			$possible_types[ $allowed_post_type ] = Types::post_object( $allowed_post_type );
+class PostObjectUnion {
+
+	/**
+	 * Registers the Type
+	 *
+	 * @param TypeRegistry $type_registry
+	 *
+	 * @access public
+	 * @return void
+	 */
+	public static function register_type( TypeRegistry $type_registry ) {
+		register_graphql_union_type(
+			'PostObjectUnion',
+			[
+				'name'        => 'PostObjectUnion',
+				'typeNames'   => self::get_possible_types(),
+				'resolveType' => function( $value ) use ( $type_registry ) {
+
+					$type = null;
+					if ( isset( $value->post_type ) ) {
+						$post_type_object = get_post_type_object( $value->post_type );
+						if ( isset( $post_type_object->graphql_single_name ) ) {
+							$type = $type_registry->get_type( $post_type_object->graphql_single_name );
+						}
+					}
+
+					return ! empty( $type ) ? $type : null;
+				},
+			]
+		);
+	}
+
+	/**
+	 * Returns a list of possible types for the union
+	 *
+	 * @access public
+	 * @return array
+	 */
+	public static function get_possible_types() {
+		$possible_types     = [];
+		$allowed_post_types = \WPGraphQL::get_allowed_post_types();
```

Comment:
Change `\WPGraphQL::get_allowed_post_types();` to `get_post_types( [ 'show_in_graphql' => true ] );` so custom post-types registered with a custom definition can still be resolved.

Reply:
@kidunot89 it looks like we use the `\WP_GraphQL::get_allowed_post_types()` several times throughout the plugin, and it calls `get_post_types( [ 'show_in_graphql' => true ])` then filters and applies another filter. Is there an issue with that? Is the static method problematic or something?

Reply:
Happy to change, just want to make sure I understand